Modules/System/SharePoint/src/helpers/SharePointRequestManager.Codeunit.al

```diff
@@ -0,0 +1,94 @@
+    var
+        SharePointHttpContent: Codeunit "SharePoint Http Content";
+
+    begin
+        OperationResponse := SendRequest(PrepareRequestMsg("Http Request Type"::POST, SharePointUriBuilder, SharePointHttpContent));
+    end;
+
+    procedure Post(SharePointUriBuilder: Codeunit "SharePoint Uri Builder"; SharePointHttpContent: Codeunit "SharePoint Http Content") OperationResponse: Codeunit "SharePoint Operation Response"
+    begin
+        OperationResponse := SendRequest(PrepareRequestMsg("Http Request Type"::POST, SharePointUriBuilder, SharePointHttpContent));
+    end;
+
+    [NonDebuggable]
+    local procedure PrepareRequestMsg(HttpRequestType: Enum "Http Request Type"; SharePointUriBuilder: Codeunit "SharePoint Uri Builder") RequestMessage: HttpRequestMessage
+    var
+        Headers: HttpHeaders;
+    begin
+        RequestMessage.Method(Format(HttpRequestType));
+        RequestMessage.SetRequestUri(SharePointUriBuilder.GetUri());
+        RequestMessage.GetHeaders(Headers);
+        Headers.Add('Accept', 'application/json');
+    end;
+
+    [NonDebuggable]
+    local procedure PrepareRequestMsg(HttpRequestType: Enum "Http Request Type"; SharePointUriBuilder: Codeunit "SharePoint Uri Builder"; SharePointHttpContent: Codeunit "SharePoint Http Content") RequestMessage: HttpRequestMessage
+    var
+        Headers: HttpHeaders;
+        HttpContent: HttpContent;
+    begin
+        RequestMessage.Method(Format(HttpRequestType));
+        RequestMessage.SetRequestUri(SharePointUriBuilder.GetUri());
+
+        RequestMessage.GetHeaders(Headers);
+        Headers.Add('Accept', 'application/json;odata=verbose');
+
+        if SharePointHttpContent.GetContentLength() > 0 then begin
+            HttpContent := SharePointHttpContent.GetContent();
+            HttpContent.GetHeaders(Headers);
+
+            if Headers.Contains('Content-Length') then
+                Headers.Remove('Content-Length');
+            Headers.Add('Content-Length', Format(SharePointHttpContent.GetContentLength()));
+
+            if Headers.Contains('Content-Type') then
+                Headers.Remove('Content-Type');
+
+            if SharePointHttpContent.GetContentType() <> '' then
+                Headers.Add('Content-Type', SharePointHttpContent.GetContentType());
+
+            Headers.Add('X-RequestDigest', SharePointHttpContent.GetRequestDigest());
+            RequestMessage.Content(HttpContent);
+        end;
+
+    end;
+
+    [NonDebuggable]
+    local procedure SendRequest(HttpRequestMessage: HttpRequestMessage) OperationResponse: Codeunit "SharePoint Operation Response"
+    var
+        HttpResponseMessage: HttpResponseMessage;
+    begin
+        Authorization.Authorize(HttpRequestMessage);
+
+        if not HttpClient.Send(HttpRequestMessage, HttpResponseMessage) then
+            Error(OperationNotSuccessfulErr);
+
+        if not HttpResponseMessage.IsSuccessStatusCode() then
```

Comment:
SharePoint is using throttling and it is strongly recommended to handle status code 429 in a proper way.
For more info see: https://docs.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online

Reply:
Added better error handling.
Whoever uses the library will be able to retrieve the retry-after value for a failed request and act accordingly.
I've opted not to retry automatically.